commit 5c8d5e337b9359be4df99c003d71028fc06ef751
Author: György Balló <ballogy@freestart.hu>
Date:   Tue Oct 18 10:50:21 2011 +0200

    Autostart only in GNOME and Unity
    
    Don't show desktop icons in other DEs, e.g Xfce and KDE uses it's own desktop implementation.

diff --git a/data/nautilus-autostart.desktop.in b/data/nautilus-autostart.desktop.in
index f105b7b..7252f87 100644
--- a/data/nautilus-autostart.desktop.in
+++ b/data/nautilus-autostart.desktop.in
@@ -2,4 +2,5 @@
 Type=Application
 Name=Files
 Exec=nautilus -n
+OnlyShowIn=GNOME;Unity;
 AutostartCondition=GSettings org.gnome.desktop.background show-desktop-icons
